### PR TITLE
MM-20653: Add role drop-down to to chat-facing team groups

### PR DESCRIPTION
**Pull request by catalintomai**
#### Summary
Add role drop-down to to chat-facing team groups

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20653

#### Related Pull Requests
- Has webapp changes https://github.com/mattermost/mattermost-webapp/pull/4490
- Has redux changes https://github.com/mattermost/mattermost-redux/pull/1009

**Note: it is superseded by https://github.com/mattermost/mattermost-server/pull/13361**

### CODE REVIEW

**Comment by catalintomai** [comment]
> I think we should close this PR in favour of #13361 which has these changes plus all of the other required backend changes, tests the store changes, and does not include the `include_scheme_admin` query parameter—which I don't think is necessary because a) it's a non-breaking addition to an API response payload, and b) the query is no slower with the included field.

Thanks, I don't have a strong opinion on this, mostly because it is a groupsBySyncableBaseQuery() method, which the way I read it, it returns group info in the context of a syncable relationship and SchemeAdmin is a syncable property. Also, if you reduce the semantics of "opts" to only parameterize the query for things that are extraneous/computed on top of the existing table(s) column values(e.g. member count), then it fits. 
That being said, I still believe that parameterizing the query even for existing group/syncable table columns could be useful, perhaps even needed in the future, especially in the light of your point b) above -  if instead of a T/F value we would have had a larger value (e.g. large string, etc.) then I surmise we prob would not want to always return the column, unless explicitly asked for. Which raises the question - where would be threshold be for not always returning the column value and instead, parameterize the query.

File: api4/group.go [hunk 1]
```diff
@@ -337,8 +337,9 @@ func patchGroupSyncable(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if !c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_MANAGE_SYSTEM) {
-		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+	appErr := verifyLinkUnlinkPermission(c, syncableType, syncableID)
+	if appErr != nil {
+		c.Err = appErr
 		return
 	}
```
Comment (catalintomai) [inline api4/group.go]: Martin, are you going to integrate these changes into your PR then?

**Comment by hahmadia** [comment]
@catalintomai Am I able to remove myself from the PR as a reviewer since it seems like we are going in the direction of using Martin's PR changes?

**Comment by catalintomai** [comment]
> @catalintomai Am I able to remove myself from the PR as a reviewer since it seems like we are going in the direction of using Martin's PR changes?

Sure, I'll remove you, will close this PR once Martin's changes go in.

**Comment by jasonblais** [comment]
@catalintomai I added a 'do not merge' label since it sounds like we'll eventually close this one in favor of https://github.com/mattermost/mattermost-server/pull/13361

Please correct me if I misunderstood!

File: api4/group.go (hunk 1 above)
Comment (mkraft) [reply api4/group.go]: @catalintomai Yes I have integrated it in. https://github.com/mattermost/mattermost-server/pull/13361/commits/9e6dedde15bfc6d01ac1a9518ff8a792a0357022. Thanks.

**Pull request closed without merging**